checker/src/types/calling.rs

```diff
@@ -820,10 +820,17 @@ fn call_logical<B: CallCheckingBehavior>(
 						top_environment,
 						types,
 					);
-					let reflects_dependency = if types.get_type_by_id(result_as_type).is_constant()
-					{
-						None
-					} else {
+
+					let should_create_dependent_output = !types
```

Comment:
@lemueldls found the fix for `function newRegExp(name: string) { return new RegExp(name) }`. Turns out for `new RegExp` with some non-constant string it creates a new object with regexp prototype (bc it is a constructor), which I think is fine. But the problem is because it is just an object, it is considered constant and so didn't generate a non-constant result for the function call. Now fixed by changing adding this condition to account for the edge case.

Reply:
Would be cool to get `String.prototype.match` and `String.prototype.matchAll` etc working but need to do some investigation into all the oddities of the `Symbol.match` stuff

Also just sent you an invite to the contributor and sponsor repository that contains the LSP if you want to try it out ([like I did earlier](https://x.com/kaleidawave/status/1828440978992025937). Repository here https://github.com/eznolab/super-ezno-and-lsp